src/rules/shared.ts

```diff
@@ -97,51 +97,39 @@ export const getJsonObjectFromPath = (json: string, path: string) => {
   return get(safeJsonParse(json), path)
 }
 
-export const setJsonObjectFromPath = (
-  json: string,
-  path: string,
-  value: string
-) => {
+const setJsonObjectFromPath = (json: string, path: string, value: string) => {
   const jsonObject = safeJsonParse(json)
   set(jsonObject ?? {}, path, value)
   return JSON.stringify(jsonObject)
 }
 
-export const replaceContent = (
-  content: string | null,
-  value: string,
+export const replaceBeginEndPattern = (
+  content: string,
+  begin: string,
+  end: string,
   newValue: string
 ) => {
-  return content?.replaceAll(value, newValue) ?? null
-}
-
-export const replaceUrl = (url: string, value: string, newValue: string) => {
-  return url.replaceAll(value, newValue)
-}
-
-export const replaceHeaders = (
-  headers: Header[],
-  value: string,
-  newValue: string
-): Header[] => {
-  return headers.map(([key, headerValue]) => {
-    const replacedValue = headerValue.replaceAll(value, newValue)
-    return [key, replacedValue]
-  })
+  const regex = new RegExp(`${escapeRegExp(begin)}(.*?)${escapeRegExp(end)}`)
+  return content.replace(regex, `${begin}${newValue}${end}`)
 }
 
-export const replaceCookies = (
-  cookies: Cookie[],
-  value: string,
+export const replaceRegexPattern = (
+  content: string,
+  regexString: string,
   newValue: string
-): Cookie[] => {
-  return cookies.map(([key, cookieValue]) => {
-    const replacedValue = cookieValue.replaceAll(value, newValue)
-    return [key, replacedValue]
+) => {
+  const regex = new RegExp(regexString)
+  return content.replace(regex, (fullMatch, ...groups) => {
+    // If there's a capture group, replace only that group
+    if (groups[0] !== undefined) {
+      return fullMatch.replace(groups[0] as string, newValue)
+    }
+    // If no capture group, replace the entire match
+    return newValue
   })
 }
 
-export const replaceBeginEndBody = (
+const replaceBeginEndBody = (
   selector: BeginEndSelector,
   request: Request,
   variableName: string
```

Comment:
Here, `variableName` does not look like the correct param name (it only does when this is used for correlation). This is out of scope of this PR, but since I've already made so many changes here, should I rename it to `value`?